python/packages/nisar/workflows/gcov_runconfig.py

```diff
@@ -16,6 +17,55 @@ def load(self):
         '''
         Load GCOV specific parameters.
         '''
+        warning_channel = journal.warning('GCOVRunConfig.load()')
+
+        flag_fullcovariance = self.cfg['processing']['input_subset'][
+            'fullcovariance']
+
+        # Handle the case in which the `fullcovariance` flag field is empty.
+        if flag_fullcovariance is None:
+
+            # The choice on whether to process in full-covariance mode
+            # or not will depend on the list of frequency and polarizations
+            # to process. This list has already been been verified based
+            # on the runconfig and available RSLC polarimetric channels
+            freq_pols_dict = self.cfg['processing']['input_subset'][
+                'list_of_frequencies']
+
+            # `fullcovariance` is set to `True` if any frequency
+            # to be processed includes a full-pol dataset,
+            # i.e., includes both co-pols HH and VV
+            # and at least one cross-pol HV or VH.
+            flag_fullcovariance = False
+            for freq, pol_list in freq_pols_dict.items():
+
+                # Verify if frequency to process is full-pol. It's considered
+                # full-pol if it contains both co-pols and at least one
+                # cross-pol.
+                if (('HH' in pol_list) and ('VV' in pol_list) and
+                   ('HV' in pol_list or 'VH' in pol_list)):
+                    warning_channel.log(
+                        'The `fullcovariance` field is not specified in the'
+                        'runconfig. By default, it is set to `True` if any'
+                        'processed frequency includes a full-pol dataset,'
+                        ' and `False` otherwise.'
+                        f'Since frequency {freq} contains polarizations'
+                        f'{pol_list}, `fullcovariance` is set to `True`.')
```

Comment:
Looks like there are some spaces missing between these lines

```suggestion
                        'The `fullcovariance` field is not specified in the'
                        ' runconfig. By default, it is set to `True` if any'
                        ' processed frequency includes a full-pol dataset,'
                        ' and `False` otherwise.'
                        f' Since frequency {freq} contains polarizations'
                        f' {pol_list}, `fullcovariance` is set to `True`.')
```

Reply:
Yes! Thank you, @gmgunter !